src/core/utils/relationutils.cpp

```diff
@@ -44,3 +44,20 @@ QgsRelation RelationUtils::resolveReferencingRelation( QgsProject *project, QgsV
     return relations.first();
   return QgsRelation();
 }
+
+QgsRelation RelationUtils::createRelation( const QgsProject *project, const QgsVectorLayer *parentLayer, const QString &parentFieldName, const QgsVectorLayer *childLayer, const QString &childFieldName )
```

Comment:
As confusing as it maybe be, I would do the following changes:
- parentLayer -> referencedLayer 
- parentFieldName -> referencedFieldName
- childLayer -> referencingLayer
- childFieldName -> referencingFieldName

That's be less confusing on the long run. Feel invited to add some context in the documentation so people know what referenced and referencing means.